### PR TITLE
[WIP DNM] Standard2.0

### PR DESCRIPTION
**Pull request by Jlalond**
#130 

So far this has been a somewhat dirty hack to convert it, as I have no idea what I'm doing.

Magically 90% of the unit tests are working as of making this PR, but I believe some of the non-integration tests need to be fixed, or deleted.

@damieng  some of the NArgs stuff is no longer existing, should I custom tool the same support or refactor the tests? I'm anticipating you suggest refactoring but you are the maintainer 👍

### CODE REVIEW

File: Source/ElasticLINQ.ElasticsearchNet/ElasticLINQ.ElasticsearchNet.csproj [hunk 1]
```diff
@@ -1,79 +1,19 @@
...
   <ItemGroup>
-    <ProjectReference Include="..\ElasticLINQ\ElasticLINQ.csproj">
-      <Project>{875e1599-5b7f-4084-bbe0-0d70c3c80428}</Project>
-      <Name>ElasticLINQ</Name>
-    </ProjectReference>
-  </ItemGroup>
-  <ItemGroup>
-    <None Include="packages.config" />
+    <Reference Include="Elasticsearch.Net">
+      <HintPath>..\..\..\..\Users\Jlalonde1\.nuget\packages\elasticsearch.net\7.4.0\lib\netstandard2.0\Elasticsearch.Net.dll</HintPath>
```
Comment (damieng) [inline Source/ElasticLINQ.ElasticsearchNet/ElasticLINQ.ElasticsearchNet.csproj]: This package ref hint is specific to your machine.

File: Source/ElasticLINQ.Test/Test/TestableElasticQueryTests.cs [hunk 2]
```diff
@@ -52,7 +52,7 @@ public static void ExplicitGetEnumeratorCapturesQueryOnContext()
 
             var dummy = ((IEnumerable) new TestableElasticQuery<FakeClass>(context).Where(f => f.Name == "a")).GetEnumerator();
 
-            Assert.Equal(1, context.Requests.Count);
+            Assert.Single(context.Requests);
```
Comment (damieng) [inline Source/ElasticLINQ.Test/Test/TestableElasticQueryTests.cs:55]: Is this general clean up or were the tests failing with the old logic?

File: Source/ElasticLINQ/ElasticLINQ.csproj [hunk 3]
```diff
@@ -1,166 +1,13 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
-<Project ToolsVersion="4.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
-  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
+﻿<Project Sdk="Microsoft.NET.Sdk.Web">
+
   <PropertyGroup>
-    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
-    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
-    <ProjectGuid>{875E1599-5B7F-4084-BBE0-0D70C3C80428}</ProjectGuid>
```
Comment (damieng) [inline Source/ElasticLINQ/ElasticLINQ.csproj:7]: How are you converting these csproj files?  I'm not keen on losing the ProjectGuid as it will mess up people upgrading, strong name signing, internals visible to...

File: Source/ElasticLINQ/ElasticLINQ.csproj [hunk 4]
```diff
@@ -1,166 +1,13 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
-<Project ToolsVersion="4.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
-  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
+﻿<Project Sdk="Microsoft.NET.Sdk.Web">
```
Comment (damieng) [inline Source/ElasticLINQ/ElasticLINQ.csproj]: I think this should be `<Project Sdk="Microsoft.NET.Sdk">`

File: Source/ElasticLINQ.ElasticsearchNet.Test/ElasticLINQ.ElasticsearchNet.Test.csproj [hunk 5]
```diff
@@ -1,117 +1,28 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
-<Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
-  <Import Project="..\packages\xunit.runner.visualstudio.2.2.0\build\net20\xunit.runner.visualstudio.props" Condition="Exists('..\packages\xunit.runner.visualstudio.2.2.0\build\net20\xunit.runner.visualstudio.props')" />
-  <Import Project="..\packages\xunit.runner.msbuild.2.1.0\build\portable-net45+win8+wp8+wpa81\xunit.runner.msbuild.props" Condition="Exists('..\packages\xunit.runner.msbuild.2.1.0\build\portable-net45+win8+wp8+wpa81\xunit.runner.msbuild.props')" />
-  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
+﻿<Project Sdk="Microsoft.NET.Sdk.Web">
```
Comment (damieng) [inline Source/ElasticLINQ.ElasticsearchNet.Test/ElasticLINQ.ElasticsearchNet.Test.csproj]: `<Project Sdk="Microsoft.NET.Sdk">`

File: Source/ElasticLINQ.ElasticsearchNet/ElasticLINQ.ElasticsearchNet.csproj [hunk 6]
```diff
@@ -1,79 +1,19 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
-<Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
-  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
+﻿<Project Sdk="Microsoft.NET.Sdk.Web">
```
Comment (damieng) [inline Source/ElasticLINQ.ElasticsearchNet/ElasticLINQ.ElasticsearchNet.csproj]: `<Project Sdk="Microsoft.NET.Sdk">`

File: Source/ElasticLINQ.IntegrationTest/ElasticLINQ.IntegrationTest.csproj [hunk 7]
```diff
@@ -1,114 +1,26 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
-<Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
-  <Import Project="..\packages\xunit.runner.visualstudio.2.2.0\build\net20\xunit.runner.visualstudio.props" Condition="Exists('..\packages\xunit.runner.visualstudio.2.2.0\build\net20\xunit.runner.visualstudio.props')" />
-  <Import Project="..\packages\xunit.runner.msbuild.2.1.0\build\portable-net45+win8+wp8+wpa81\xunit.runner.msbuild.props" Condition="Exists('..\packages\xunit.runner.msbuild.2.1.0\build\portable-net45+win8+wp8+wpa81\xunit.runner.msbuild.props')" />
-  <Import Project="..\packages\xunit.core.2.0.0\build\Xamarin.iOS\xunit.core.props" Condition="Exists('..\packages\xunit.core.2.0.0\build\Xamarin.iOS\xunit.core.props')" />
-  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
+﻿<Project Sdk="Microsoft.NET.Sdk.Web">
```
Comment (damieng) [inline Source/ElasticLINQ.IntegrationTest/ElasticLINQ.IntegrationTest.csproj]: `<Project Sdk="Microsoft.NET.Sdk">`

**Review by damieng — COMMENTED** [review]
Great start, a few minor tweaks around the project files.  I'll see if I can take a stab at compiling it and running the tests this coming weekend.

**Comment by Jlalond** [comment]
@damieng cleaned them up. I admit I blindly copied one and go started, if you know how to tell resharper to stop trying to reference local packages too that would be great, otherwise I can go back to harassing jetbrains.

**Comment by damieng** [comment]
Let's keep general clean-up separate from the .NET Standard port for now.  Helps us understand what's what.

**Comment by Jlalond** [comment]
Agreed, only tidied up your previous csproj comments.

File: Source/ElasticLINQ/ElasticLINQ.csproj (hunk 3 above)
Comment (Jlalond) [reply Source/ElasticLINQ/ElasticLINQ.csproj:7]: So originally, I actually just deleted the csproj and used the template on MSDN, I'm okay with adding back in the project guids if you'd like.

File: Source/ElasticLINQ.Test/Test/TestableElasticQueryTests.cs (hunk 2 above)
Comment (Jlalond) [reply Source/ElasticLINQ.Test/Test/TestableElasticQueryTests.cs:55]: Unintended cleanup. I used a lot of my resharper binds to delete the relink the csprojs and in doing so some of these got changed. I can totally revert these if you want to reduce the noise.

**Comment by damieng** [comment]
How you doing on this?

**Comment by Jlalond** [comment]
@damieng Hey! Sorry I misunderstood, you mentioned you were going to pull the code down (to look at the unit test I mentioned), and I got wrapped up with the holiday.

 I can get back to working on this probably this weekend, currently we're building with a few tests failing mostly because `Args.Any<>` wasn't compiling, (see [Line 182](https://github.com/ElasticLINQ/ElasticLINQ/pull/132/files#diff-a5b910c2f35889af848ba0b1089e7f35R182) )

**Comment by Jlalond** [comment]
@damieng Update:

Aside from integration tests that I cannot verify, we have quite a few tests failing and wanted to reach out for some guidance.
`LogsDebugMessagesDuringExecution` and `NonSuccessfulHttpRequestThrows` in `ElasticSearchNet` are failing because I can't get nsub to actually return a non-null obj. I'm going to dedicate myself to figuring this out (brute force or otherwise)

However, in the actually querying building part, we have quite a few failing, and I noticed the build was failing prior to this date, so I wonder if this is acceptable.

ElasticQueryExtensions:
`OrderByScoreDescendingIsAddedToExpressionTree` 
`OrderByScoreIsAddedToExpressionTree`
`ThenByScoreDescendingIsAddedToExpressionTree`
`ThenByScoreIsAddedToExpressionTree`
these are returning an empty set when it's not expected, this sounds actually broken to me. I read earlier you can't support order by's anymore, so looking for confirmation.

ElasticQueryTranslationWhereTests:
`AndOrQueryGeneratesBoolQueryWithAndArgs`
`QueryGeneratesQueryCriteria`
`QueryStringWithQueryCombinesToBoolQueryCriteria`
 expected bool criteria, got and criteria.


`BooleanConstantsGeneratesBoolMatchAllQueryAndNotMatchAllQuery` expected bool criteria, got range criteria

`BooleanFalseGeneratesNotMatchAllQuery`
`EvaluatedFalseGeneratesNotMatchAllQuery`
 Not, was Constant


`BooleanTrueGeneratesMatchAllQuery` 
`EvaluatedTrueGeneratesMatchAllQuery`
Got null, looking in this now.

`QueryAndWhereGeneratesQueryAndFilterCriteria` regexp => exists

And that's about it, we're currently building, I think this is going great so far!

**Comment by Jlalond** [comment]
@damieng Happy new year, have you been able to give me any insight on some of the tests I mentioned?

**Comment by damieng** [comment]
I'm finding it very hard to follow this as it's quite a heavy-handed upgrade.  We really need to bring back the original guids and strong-name signing.  Wiping out the existing csproj files entirely really messes up a bunch of things and adds to the noise.

Also need to remove the resharper refactorings - could potentially do that later but really need this PR to be just the essentials to convert the project over so we can keep track of what changed and why plus dig into the actual functional breaks and what could have caused them.

**Comment by Jlalond** [comment]
@damieng I understand, I definitely cobbled this together.

I'll get the csproj's fixed and I'll do my best to manually undo any non-code/style changes.

**Comment by damieng** [comment]
Now you have a compiling branch you might find it easier to create another
branch and transplant what you need across one piece at a time - e.g. first
the small parts of .csproj to get it to change target etc.

I find that way works well myself.

**Comment by Jlalond** [comment]
Temporarily closing until I prioritize myself better (and finish moving)
